### DISCUSSION TITLE
how to calculate nginx plus ingress controller instance?

### DISCUSSION THREAD

**Discussion by nguyenphu123** [nginx/kubernetes-ingress#6558] [Q&A] [answered]
Hello, I have deployed nginx plus ingress controller on a kubernetes cluster. Is there any doc about calculating the instances I am using ? Or a formula to calculate the instances need for a kubernetes system ?

**Comment by jjngx** [comment]
Hi @nguyenphu123, could you please describe your use case? The number of instances you declare in a deployment obj (number of replicas), for example: [link](https://github.com/nginxinc/kubernetes-ingress/blob/e1dc26850b576c3dc1f643dda0eca692e61633e8/deployments/deployment/nginx-plus-ingress.yaml#L7)

**Reply by nguyenphu123** [reply]
My nginx ingress controller currently running with 2 pods and 2 replicas. I also have 2 services with 2 pods and 2 replicas each that are running behind the ingress controller.

**Comment by vepatel** [comment, marked as answer]
Hi @nguyenphu123 there's no set formula to select number of replicas/pods but the general guideline is to use horizontal pod scaling to match the demand. Have a look at https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/
To apply this to NIC, you can use `controller.autoscaling...` helm parameters https://docs.nginx.com/nginx-ingress-controller/installation/installing-nic/installation-with-helm/#configuration
